config/checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,199 @@
+        <module name="PackageDeclaration"/>
+        <!--		<module name="PackageName">-->
+        <!--			&lt;!&ndash; require a package following the following structure:-->
+        <!--			base package name: net.fabricmc.fabric-->
+        <!--			+ api/implementation/mixin subpackage: api/impl/mixin-->
+        <!--			+ client/dedicated server/common env subpackage: client/server/<nothing>-->
+        <!--			+ module name subpackage, singular, may contain multiple .-separated parts-->
+        <!--			+ api only: module major version with v prefix (e.g. v1)-->
+        <!--			+ other subpackages as needed, all singular-->
+        <!--			The regex works as follows:-->
+        <!--			It matches (=succeeds) for one of these cases:-->
+        <!--			- net.fabricmc.fabric.api.client.<module-name>.v<version>[.<extra packages...>]-->
+        <!--			- net.fabricmc.fabric.api.server.<module-name>.v<version>[.<extra packages...>]-->
+        <!--			- net.fabricmc.fabric.api.<module-name>.v<version>[.<extra packages...>]-->
+        <!--			- net.fabricmc.fabric.(impl|mixin).client.<module-name>[.<extra packages...>]-->
+        <!--			- net.fabricmc.fabric.(impl|mixin).server.<module-name>[.<extra packages...>]-->
+        <!--			- net.fabricmc.fabric.(impl|mixin).<module-name>[.<extra packages...>]-->
+        <!--			- <any legacy package>-->
+        <!--			where <module-name> is a set of '.'-separated words, all in singular (not ending with s except for ss) and not starting with client. or server.-->
+        <!--			and <version> is a positive integer (1, 2, 3, ...)-->
+        <!--			and <extra packages...> is a set of '.'-separated words with all the first potentially containing digits.-->
+        <!--			Negative lookahead ensures that client/server can't be replaced with common disguised as the module name.-->
+        <!--			The regex is implemented in 3 parts:-->
+        <!--			- the net.fabricmc.fabric. prefix-->
+        <!--			- patterns for-->
+        <!--				- api packages: api + <not common> + client/server/nothing + <module-name> + 'v' + <version>-->
+        <!--				- impl+mixin packages : impl/mixin + <not common> + client/server/nothing + <module-name>-->
+        <!--				- literal legacy packages (exceptions)-->
+        <!--			- largely unconstained trailing subpackages-->
+        <!--			&ndash;&gt;-->
+        <!--			<property name="format"-->
+        <!--					  value="^net\.fabricmc\.fabric\.(api(?!\.common\.)(\.client|\.server|)(\.(?!client\.|server\.)[a-z]+([a-rt-z]|ss))+\.v[1-9][0-9]*|(impl|mixin|test)(?!\.common\.)(\.client|\.server|)(\.(?!client\.|server\.)[a-z]+[a-rt-z])+|api\.(event|util|biomes\.v1|registry|client\.screen|container|block|entity|client\.itemgroup|client\.keybinding|tag|tools|client\.model|network|server|client\.render|resource|client\.texture))(|\.[a-z]+(\.[a-z0-9]+)*)$"/>-->
+        <!--		</module>-->
+
+        <!--<module name="InvalidJavadocPosition"/>-->
```

Comment:
Curious on the large comment block about Fabric package formatting

Reply:
this checkstyle is qsl's, I think that comes straight from fapi